libraries/standard/http/src/http_client.c

```diff
@@ -16,22 +45,158 @@ HTTPStatus_t HTTPClient_AddHeader( HTTPRequestHeaders_t * pRequestHeaders,
+
+    if( transportStatus < 0 )
+    {
+        IotLogErrorWithArgs( "Error in sending the HTTP headers over the transport "
+                             "interface. Transport status %d.",
+                             transportStatus );
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+    else if( transportStatus != pRequestHeaders->headersLen )
+    {
+        IotLogErrorWithArgs( "Attempted to send %d of HTTP headers, transport "
+                             " reported it sent %d.",
+                             pRequestHeaders->headersLen,
+                             transportStatus );
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t _sendHttpBody( const HTTPTransportInterface_t * pTransport,
+                                   const uint8_t * pRequestBodyBuf,
+                                   size_t reqBodyBufLen )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    int32_t transportStatus = 0;
+
+    /* Send the HTTP body over the network. */
+    if( pRequestBodyBuf != NULL )
+    {
+        transportStatus = pTransport->send( pTransport->pContext,
+                                            pRequestBodyBuf,
+                                            reqBodyBufLen );
+
+        if( transportStatus < 0 )
+        {
+            IotLogErrorWithArgs( "Error in sending the HTTP body over the "
+                                 "transport interface. Transport status %d.",
+                                 transportStatus );
+            returnStatus = HTTP_NETWORK_ERROR;
+        }
+        else if( transportStatus != reqBodyBufLen )
+        {
+            IotLogErrorWithArgs( "Attempted to send %d of HTTP body, transport "
+                                 "reported it sent %d.",
+                                 reqBodyBufLen,
+                                 transportStatus );
+            returnStatus = HTTP_NETWORK_ERROR;
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t _receiveHttpResponse( const HTTPTransportInterface_t * pTransport,
+                                          HTTPResponse_t * pResponse )
+{
+    /* TODO: Receive the HTTP response with parsing. */
+    return HTTP_SUCCESS;
+}
+
+/*-----------------------------------------------------------*/
+
 HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t * pTransport,
                               const HTTPRequestHeaders_t * pRequestHeaders,
                               const uint8_t * pRequestBodyBuf,
                               size_t reqBodyBufLen,
                               HTTPResponse_t * pResponse )
 {
-    return HTTP_NOT_SUPPORTED;
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+
+    if( pTransport == NULL )
+    {
+        IotLogError( "The transport interface cannot be NULL." );
```

Comment:
Please add the _failure event/outcome_ before the _cause_ of the event in the logs.
Each of these should contain: "Failed to send HTTP request: <specific input parameter> is invalid:"

Reply:
So what would this be? "Transport interface is null: failure."?